float64 px
float64 py
float64 radius